x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java

```diff
@@ -271,22 +273,25 @@ protected void onStart(long now, ActionListener<Boolean> listener) {
             return;
         }
 
-        SetOnce<Map<String, String>> deducedDestIndexMappings = new SetOnce<>();
-
-        ActionListener<Void> finalListener = ActionListener.wrap(r -> {
-            try {
-                // if we haven't set the page size yet, if it is set we might have reduced it after running into an out of memory
-                if (context.getPageSize() == 0) {
-                    configurePageSize(getConfig().getSettings().getMaxPageSearchSize());
-                }
+        if (context.getAuthState() != null && HealthStatus.RED.equals(context.getAuthState().getStatus())) {
+            // AuthorizationState status is RED which means there was permission check error during PUT or _update.
+            listener.onFailure(
+                new ElasticsearchSecurityException(
+                    TransformMessages.getMessage(TransformMessages.TRANSFORM_CANNOT_START_WITHOUT_PERMISSIONS, getConfig().getId())
```

Comment:
Should we incorporate the `authState.getLastAuthError()` into this error message?

Reply:
We could, it makes it look like this:
```
issues=[
  {
    issue=Privileges check failed,
    count=1,
    details=Cannot create transform [transform-permissions-defer-unattended] because user john_junior lacks the required permissions [transform-permissions-defer-unattended-dest:[create_index, index, read], transform-permissions-defer-unattended-index:[read, view_index_metadata]],
    type=privileges_check_failed
  },
  {
    issue=Transform indexer failed,
    first_occurrence=1711472840615,
    count=3,
    details=org.elasticsearch.ElasticsearchSecurityException: Cannot start transform [transform-permissions-defer-unattended] because user lacks required permissions, more details: [Cannot create transform [transform-permissions-defer-unattended] because user john_junior lacks the required permissions [transform-permissions-defer-unattended-dest:[create_index, index, read], transform-permissions-defer-unattended-index:[read, view_index_metadata]]],
    type=transform_indexer_failed
  }
]
```

It always dupliactes the above message, which is probably fine, but that's why the first iteration is "check the `privileges_check_failed` message"

Reply:
Ok. If the `"check the privileges_check_failed message"` is always there then leave it as it was.